internal/pkg/deploy/sns.go

```diff
@@ -0,0 +1,65 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package deploy holds the structures to deploy infrastructure resources.
+// This file defines workload deployment resources.
+package deploy
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+var errInvalidTopicARN = errors.New("ARN is not a Copilot SNS topic")
+var errInvalidARN = errors.New("invalid ARN format")
+var errInvalidARNService = errors.New("ARN is not an SNS topic")
+
+type Topic struct {
+	ARN  string
+	App  string
+	Env  string
+	Wkld string
+}
+
+// Name returns the name of the given SNS topic, stripped of its "app-env-wkld" prefix.
+func (t Topic) Name() (string, error) {
+	parsedARN, err := t.parse()
+	if err != nil {
+		return "", err
+	}
+	prefix := fmt.Sprintf(fmtSNSTopicNamePrefix, t.App, t.Env, t.Wkld)
```

Comment:
I'm wondering if checking whether the topic is copilot-managed should be the responsibility of `Name()` or not 🤔  I feel like a function named `Name()`, should just return the name, without any additional assumption on the topic. Does the usage of the function requires this validation?